packages/material-ui/src/TablePagination/TablePagination.js

```diff
@@ -237,14 +243,39 @@ TablePagination.propTypes = {
    *
    * @param {object} event The event source of the callback.
    * @param {number} page The page selected.
+   *
+   * @deprecated Will be removed in v5. Use the onPageChange prop instead.
    */
-  onChangePage: PropTypes.func.isRequired,
+  onChangePage: chainPropTypes(PropTypes.func, () => {
+    return new Error(
+      'Warning: Failed props type: Material-UI onChangePage was deprecated. Use onPageChange instead',
+    );
+  }),
+  /**
+   * Callback fired when the page is changed.
+   *
+   * @param {object} event The event source of the callback.
+   * @param {number} page The page selected.
+   */
+  onPageChange: PropTypes.func.isRequired,
+  /**
+   * Callback fired when the number of rows per page is changed.
+   *
+   * @param {object} event The event source of the callback.
+   *
+   * @deprecated Will be removed in v5. Use the onRowsPerPageChange prop instead.
+   */
+  onChangeRowsPerPage: chainPropTypes(PropTypes.func, () => {
+    return new Error(
+      'Warning: Failed props type: Material-UI onChangeRowsPerPage was deprecated. Use onRowsPerPageChange instead',
+    );
```

Comment:
`deprecatedPropType` already covers this use case, with the advantage to update the generated API page.

Reply:
Wasn't aware of this. Updated, looks better.